src/compiler/scala/tools/nsc/classpath/AggregateClassPath.scala

```diff
@@ -72,7 +73,13 @@ case class AggregateClassPath(aggregates: Seq[ClassPath]) extends ClassPath {
     getDistinctEntries(_.sources(inPackage))
 
   override private[nsc] def list(inPackage: String): ClassPathEntries = {
-    val (packages, classesAndSources) = aggregates.map(_.list(inPackage)).unzip
+    val (packages, classesAndSources) = aggregates.map { cp =>
+      try {
+        cp.list(inPackage)
+      } catch {
+        case ex: java.io.IOException => throw new FatalError(ex.getMessage)
```

Comment:
This discards potentially important information (the stack trace and class of `ex`).